Update firmware it's not working

**Describe the bug**
Trying to update the firmware trough bootloader it's  not working.
```
go run cmd/cli/cli.go firmwareUpdate -f SkyCoin/hardware-wallet/tiny-firmware/skycoin.bin 
File : SkyCoin/hardware-wallet/tiny-firmware/skycoin.bin
Hash: 0d89e2195ea374f0191d48a7e8c475224fa0efe359566a8f20e4e86ef0b0a06b
[2019-04-03T16:02:54-04:00] INFO [device-wallet]: Length of firmware 211992
(*devicewallet.Device)(0xc00008d890)({
 Driver: (*devicewallet.Driver)(0xc000014a50)({
  deviceType: (devicewallet.DeviceType) USB
 }),
 dev: (io.ReadWriteCloser) <nil>,
 simulateButtonPress: (bool) false,
 simulateButtonType: (devicewallet.ButtonType) -1
})
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x6e05ac]

goroutine 1 [running]:
github.com/skycoin/hardware-wallet-go/src/device-wallet.sendToDevice(0x0, 0x0, 0xc00015f1c0, 0x1, 0x1, 0xa4f100, 0x7eb460, 0xc00000e018, 0xc0001178f0, 0x1, ...)
        /usr/share/gocode/src/github.com/skycoin/hardware-wallet-go/src/device-wallet/helper.go:102 +0xfc
github.com/skycoin/hardware-wallet-go/src/device-wallet.(*Driver).SendToDevice(0xc000014a50, 0x0, 0x0, 0xc00015f1c0, 0x1, 0x1, 0x0, 0x0, 0x691a48, 0xc0001178c8, ...)
        /usr/share/gocode/src/github.com/skycoin/hardware-wallet-go/src/device-wallet/helper.go:69 +0x60
github.com/skycoin/hardware-wallet-go/src/device-wallet.(*Device).FirmwareUpload(0xc00008d890, 0xc000172000, 0x33c18, 0x33e18, 0xf074a35e19e2890d, 0x2275c4e8a7481d19, 0x8f6a5659e3efa04f, 0x6ba0b0f06ee8e420, 0x0, 0x0)
        /usr/share/gocode/src/github.com/skycoin/hardware-wallet-go/src/device-wallet/device-wallet.go:314 +0x2f6
github.com/skycoin/hardware-wallet-go/src/cli.firmwareUpdate.func1(0xc0000aa420)
        /usr/share/gocode/src/github.com/skycoin/hardware-wallet-go/src/cli/firmware_update.go:39 +0x355
github.com/skycoin/hardware-wallet-go/vendor/github.com/urfave/cli.HandleAction(0x729720, 0x7adea0, 0xc0000aa420, 0xc00006c600, 0x0)
        /usr/share/gocode/src/github.com/skycoin/hardware-wallet-go/vendor/github.com/urfave/cli/app.go:492 +0x7c
github.com/skycoin/hardware-wallet-go/vendor/github.com/urfave/cli.Command.Run(0x79d076, 0xe, 0x0, 0x0, 0x0, 0x0, 0x0, 0x7a0ea5, 0x19, 0x0, ...)
        /usr/share/gocode/src/github.com/skycoin/hardware-wallet-go/vendor/github.com/urfave/cli/command.go:210 +0x9a2
github.com/skycoin/hardware-wallet-go/vendor/github.com/urfave/cli.(*App).Run(0xc0000b8d00, 0xc000010100, 0x4, 0x4, 0x0, 0x0)
        /usr/share/gocode/src/github.com/skycoin/hardware-wallet-go/vendor/github.com/urfave/cli/app.go:255 +0x687
github.com/skycoin/hardware-wallet-go/src/cli.(*App).Run(0xc0000b8d00, 0xc000010100, 0x4, 0x4, 0x0, 0x0)
        /usr/share/gocode/src/github.com/skycoin/hardware-wallet-go/src/cli/cli.go:119 +0x4b
main.main()
        /usr/share/gocode/src/github.com/skycoin/hardware-wallet-go/cmd/cli/cli.go:17 +0xb5
exit status 2
```

**Environment information:**

- OS: Any
- Platform: Any
- Firmware Version: develop

**Steps to Reproduce**
Just run `skycoin-hw-cli firmwareUpdate -f path/to/firmware`

**Actual behavior**
Just not working, unable o upload a new firmware.

**Expected behavior**
Update the firmware.

--- a/src/device-wallet/device-wallet.go
+++ b/src/device-wallet/device-wallet.go
@@ -292,15 +292,12 @@ func (d *Device) FirmwareUpload(payload []byte, hash [32]byte) error {
 	if d.Driver.DeviceType() != DeviceTypeUSB {
 		return errors.New("wrong device type")
 	}
-	var err error
-	d.dev, err = d.Driver.GetDevice()
-	if err != nil {
+	if err := d.Connect(); err != nil {
 		return err
 	}
 	defer d.dev.Close()
 
-	err = Initialize(d.dev)
-	if err != nil {
+	if err := Initialize(d.dev); err != nil {
 		return err
 	}
 

--- a/src/device-wallet/device-wallet.go
+++ b/src/device-wallet/device-wallet.go
@@ -292,13 +292,14 @@ func (d *Device) FirmwareUpload(payload []byte, hash [32]byte) error {
 	if d.Driver.DeviceType() != DeviceTypeUSB {
 		return errors.New("wrong device type")
 	}
-	dev, err := d.Driver.GetDevice()
+	var err error
+	d.dev, err = d.Driver.GetDevice()
 	if err != nil {
 		return err
 	}
-	defer dev.Close()
+	defer d.dev.Close()
 
-	err = Initialize(dev)
+	err = Initialize(d.dev)
 	if err != nil {
 		return err
 	}
@@ -332,7 +333,7 @@ func (d *Device) FirmwareUpload(payload []byte, hash [32]byte) error {
 	if err != nil {
 		return err
 	}
-	return d.Driver.SendToDeviceNoAnswer(dev, chunks)
+	return d.Driver.SendToDeviceNoAnswer(d.dev, chunks)
 }
 
 // GetFeatures send Features message to the device

